app/javascript/components/ActionItemForm/index.js

```diff
@@ -0,0 +1,338 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import validator from 'validator';
+import MUIRichTextEditor from 'mui-rte';
+import HouseIcon from '@material-ui/icons/House';
+import EcoSharpIcon from '@material-ui/icons/EcoSharp';
+import CreateSharpIcon from '@material-ui/icons/CreateSharp';
+import NoteSharpIcon from '@material-ui/icons/NoteSharp';
+import SentimentSatisfiedSharpIcon from '@material-ui/icons/SentimentSatisfiedSharp';
+import IconButton from '@material-ui/core/IconButton';
+import {
+  Avatar,
+  Button,
+  TextField,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Grid,
+} from '@material-ui/core';
+import { withStyles, MuiThemeProvider } from '@material-ui/core/styles';
+
+import { convertToRaw } from 'draft-js';
+import { styles, theme } from './styles';
+
+class ActionItemForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      description: this.props.description,
+      type: this.props.type,
+      title: this.props.title,
+      open: false,
+      dueDate: this.props.dueDate,
+      category: this.props.category,
+      errors: {
+        title: '',
+      },
+    };
+    // this.onChange = editorState => this.setState({ editorState });
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleCategoryChange = this.handleCategoryChange.bind(this);
+  }
+
+  handleOpen() {
+    this.setState({ open: true });
+  }
+
+  handleClose() {
+    this.setState({
+      open: false,
+      title: this.props.title,
+    });
+    if (this.state.type === 'edit') {
+      (this.state.title = this.props.title)(
+        (this.state.description = this.props.description),
+      );
+    }
+  }
+
+  checkErrors = field => () => {
```

Comment:
For this file, either keep functions as arrow notation like this function checkErrors or use the format as handleClose(). either are fine but keep it consistent.